### ISSUE TITLE
Change name of sigmoid_cross_entropy_with_logits to log_loss

### ISSUE THREAD

**Issue by zackchase** [tensorflow/tensorflow#699]
Hi folks. This name sigmoid_cross_entropy_with_logits struck me as strange. In addition to being a mouthful, we're calling it cross_entropy when it really is just binary cross_entropy calculated separately on every node. 

The more intuitive name I would give to this function would be log_loss or mean_log_loss (if normalized). I think this would be both easier to write, and easier to understand.

**Comment by martinwicke** [comment]
The name may not be the best one, but we probably want to avoid saying "loss" in the function (since it's use it not necessarily restricted to being used a loss).

**Comment by zackchase** [comment]
Good point. Similarly, it's not restricted to being used with sigmoid units. Maybe better to call it binary_cross_entropy?

**Comment by martinwicke** [comment]
Sounds good to me. @sherrym, I'll assign you for summary judgment.

**Comment by ppwwyyxx** [comment]
Following your points on naming, maybe a better name for `nn.l2_loss` is `nn.l2_norm`.

**Comment by goodfeli** [comment]
@zackchase , sigmoid_cross_entropy_with_logits actually is restricted to being used with sigmoid units, in the sense that it only gives you the cross entropy if you are using such units. It is the op for binary cross entropy when the binary probability is produced by applying the sigmoid function to one argument. You could imagine other ways of getting binary probabilities, like thresholding the argument to the interval [0,1], or taking softplus(x)/(1+softplus(x)) instead of sigmoid, etc.

**Comment by girving** [comment]
Closing since changing the name is too little benefit for the large amount of required work.
